Chris enemy final

SmeeDucks opened the pull request:
finished enemies and boss

SpencerP23 commented:
So a few notes after I've messed around with this for a while:

- Love the chaos I feel throughout the fight, really interacts well with the magnet mechanic too
- I don't think I like that the ball can destroy the block bullets. More often than not navigating over there causes an accidental hit, which then causes an almost instant respawn, missing the hit. I think either it does not get destroyed, or it stays destroyed for much longer would be preferred. But again something we can mess around with for sure
- The BossHit script had the player used as the ball for hits which idk if that was just debug or not, so I did not touch it save for my own testing
- I'm able to just go through the boss's holes, an invisible tilemap of collision may be helpful, which I can throw in if you want
- Some weird stuff happened when I loaded up the scene (the ball got completely removed for one), so I modified the scene (and a prefab I realized I didn't fix) a little bit in my commit

Just double check that my scene stuff didn't destory everything on your end then you can go ahead and merge this through! Thanks so much!

SmeeDucks commented:
> So a few notes after I've messed around with this for a while:
> 
> * Love the chaos I feel throughout the fight, really interacts well with the magnet mechanic too
> * I don't think I like that the ball can destroy the block bullets. More often than not navigating over there causes an accidental hit, which then causes an almost instant respawn, missing the hit. I think either it does not get destroyed, or it stays destroyed for much longer would be preferred. But again something we can mess around with for sure
> * The BossHit script had the player used as the ball for hits which idk if that was just debug or not, so I did not touch it save for my own testing
> * I'm able to just go through the boss's holes, an invisible tilemap of collision may be helpful, which I can throw in if you want
> * Some weird stuff happened when I loaded up the scene (the ball got completely removed for one), so I modified the scene (and a prefab I realized I didn't fix) a little bit in my commit
> 
> Just double check that my scene stuff didn't destory everything on your end then you can go ahead and merge this through! Thanks so much!

Yea sorry for this seems I forgot to remove some of the debugging stuff I was doing.
- increased respawn timer for blocks after being destroyed (removing this feature might be a good idea) and increased their damage
- fixed player damaging boss
- turned back on collider for trying to leave (resets player to center)
